Nov 12th 2024 Meeting Minutes (#66)
hiero-ledger/tsc: minutes/2024-11-12.md

   - Dependency and DCO Checks
4. Discussion on GitHub Runners and Infrastructure
   - Self-Hosted vs. GitHub Runners
   - Licensing and Costs of Security Tools
5. Repository Setup and Security Policies
   - Usage & Payment of Step Security and Codacy
   - Implementation of Security.md
   - OpenSSF Scorecard Integration
6. Meeting Schedule Adjustment
7. Any Other Business
   - Formation of Working Groups
8. Action Items and Next Steps
9. Adjournment

## Meeting Minutes:

**Summary:**

The Hiero Technical Steering Committee discussed several key topics, including the approval of previous meeting minutes, project transition updates, and infrastructure considerations for GitHub runners and security tools. Emphasis was placed on the need for standardized repository setups, particularly concerning security policies and dependency management. The committee agreed to adjust the meeting time and consider forming working groups to address specialized areas more effectively. Action items were assigned to ensure progress on project transitions, security implementations, and administrative adjustments.

### **1. Welcome and Meeting Protocols**

> - **Hendrik Ebbers** reminded everyone of the **Code of Conduct** and the **Antitrust Policy** of the Renux Foundation, emphasizing the importance of maintaining professionalism, especially since participants belong to different organizations.

Reviewer: ```suggestion
- **Hendrik Ebbers** reminded everyone of the **Code of Conduct** and the **Antitrust Policy** of the Linux Foundation, emphasizing the importance of maintaining professionalism, especially since participants belong to different organizations.
```

- **Second Set of Minutes:**
  - **Hendrik** raised a concern about unnecessary personal details in the minutes.
  - He suggested that such details do not add value and should be omitted to maintain professionalism.
  - **Brandon** agreed to remove the sentence and adjust future minutes accordingly.
  - **Decision:** With the agreed amendments, the minutes will be approved and merged after **Hendrik's** final review.

### **3. Project Transition Updates**

- **Hendrik Ebbers** provided an update on the project transition to Hiero:

  **a. Maintainers and Committers Assignments:**

  - No new projects have been moved since the last meeting.
  - Blocking issues identified:
    - Unclear assignments of maintainers and committers for some projects.
    - **Hendrik** created issues and assigned responsible individuals to address these gaps.
    - Follow-up required to ensure assignments are completed.

  **b. Publishing Credentials for Maven Central and npmjs:**

  - **Local Node Project:**
    - Initially believed not to be published anywhere, but it is available on npmjs.
    - Before transferring the **Mirror Node**, they need to secure access to the npmjs account.
  - **Go SDK:**

>     - Dependencies have been updated to remove reliance on the **Serum** package.

Reviewer: ```suggestion
    - Dependencies have been updated to remove reliance on the **Ethereum** package.
```

    - **Hendrik** will re-check the Go SDK dependencies after updates from LimeChain.
    - Potential for the Go SDK to be the next project to transfer, as it doesn't require artifact publishing.
  - **Credential Transfer:**
    - Necessary to move publishing credentials for Maven Central and npmjs to the Linux Foundation.
    - **Richard Bair** acknowledged the need to investigate how to transfer these credentials.
    - **Action:** **Richard** to explore the process of transferring the Maven Central and npmjs accounts to Hiero.

  **c. Dependency and DCO Checks:**

  - **Hendrik** and **Wayne** (from LFDT) have been working on cleaning up DCO (Developer Certificate of Origin) issues.
  - Most problematic commits have been resolved, except for a few with untraceable authors (e.g., commits from "localhost@ubuntu").
  - Remaining issues have been escalated to **Diane** for further action.
  - Emphasis on ensuring all dependencies are compliant and do not introduce licensing issues.

  **d. Documentation and Guidance:**

>   - A step-by-step guide for project transition has been created, available in the `hashgraph/transfer` repository.

Reviewer: ```suggestion
  - A step-by-step guide for project transition has been created, available in the `[hiero-ledger/hiero](https://github.com/hiero-ledger/hiero)` repository.
```

  - The guide currently lacks information on publishing artifacts; this will be added once processes are established.
  - **Hendrik** is documenting the roles and permissions within repositories to standardize practices.

### **4. Discussion on GitHub Runners and Infrastructure**

- **Hendrik** raised concerns about GitHub apps and plugins:

  **a. Security Tools Licensing:**

>   - Current security tools like **Two-Step Security** and **CodeCov** require enterprise licenses for features like self-hosted runners.

Reviewer: ```suggestion
  - Current security tools like **Step Security** and **Codacy** require enterprise licenses for features like self-hosted runners.
```

- **Alex** brought up the need for standardizing repository setups, specifically:

  **a. Security Notification Policies:**

  - Noted that some Hiero repositories lack a `SECURITY.md` file, which outlines procedures for reporting and handling security vulnerabilities.
  - Emphasized the importance of having clear security policies to manage notifications and escalations effectively.

  **b. Implementation Plan:**

  - **Hendrik** acknowledged the issue and mentioned an existing GitHub issue to address it.
  - Waiting for a finalized template from LFTT to standardize the `SECURITY.md` across all repositories.
  - Plan to place the `SECURITY.md` in the `.github` directory so it applies globally to all repositories without individual files.

  **c. OpenSSF Scorecard Integration:**

  - **Hendrik** discussed the Open Source Security Foundation's (OpenSSF) Scorecard, a tool for assessing security practices in repositories.
  - Implementing the scorecard involves setting up a GitHub action that runs security checks.
  - Recognized that while valuable, this integration might be postponed until after the project transition is complete due to current workload.

  **d. Action Items:**

>   - **Hendrik** to follow up with **Wayne** on the status of the `SECURITY.md` template.

Reviewer: ```suggestion
  - **Hendrik** to follow up with **Ry** on the status of the `SECURITY.md` template.
```
